### ISSUE TITLE
haproxy: OpenSSL x86_64 Montgomery squaring procedure carry propagating bug (SecAdv 20171102)

### ISSUE THREAD

**Issue by zvercodebender** [rancher/rancher#10870]
**Rancher versions:**
rancher/server: v1.6.12
rancher/agent: v1.2.7

**Infrastructure Stack versions:**
healthcheck: v0.3.1
ipsec: v0.11.7
network-services: v0.9.3
scheduler: v0.8.2
kubernetes (if applicable):

**Docker version: (`docker version`,`docker info` preferred)**

Containers: 48
 Running: 47
 Paused: 0
 Stopped: 1
Images: 25
Server Version: 17.06.1-ce
Storage Driver: overlay
 Backing Filesystem: xfs
 Supports d_type: false
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local rancher-nfs
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 6e23458c129b551d5c9871e5174f6b1b7f6d1170
runc version: 810190ceaa507aa2727d7ae6f4790c76ec150bd2
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 3.10.0-514.26.2.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 15.22GiB
Name: fang.dmz.blackpugsoftware.com
ID: T2JJ:G4EM:76GA:Q5CH:PHKH:DQZC:RDFV:6AEX:PYCJ:SVJO:4EJE:MKZ6
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 registry.blackpugsoftware.com:443
 127.0.0.0/8
Live Restore Enabled: false

WARNING: overlay: the backing xfs filesystem is formatted without d_type support, which leads to incorrect behavior.
         Reformat the filesystem with ftype=1 to enable d_type support.
         Running without d_type support will not be supported in future releases.

**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**

3.10.0-514.26.2.el7.x86_64

**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**

Bare-metal

**Setup details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
Rancher single node with two, two host clusters

**Environment Template: (Cattle/Kubernetes/Swarm/Mesos)**

Cattle

**Steps to Reproduce:**

**Results:**

I received a recent security scan noticed of a CVE on my rancher clusters.  It turns out there is a weakness in the openssl version that is currently used for haproxy (rancher/lb-service-haproxy:v0.7.15).

https://www.openssl.org/news/secadv/20171102.txt

bn_sqrx8x_internal carry bug on x86_64 (CVE-2017-3736)
======================================================

Severity: Moderate

There is a carry propagating bug in the x86_64 Montgomery squaring procedure. No
EC algorithms are affected. Analysis suggests that attacks against RSA and DSA
as a result of this defect would be very difficult to perform and are not
believed likely. Attacks against DH are considered just feasible (although very
difficult) because most of the work necessary to deduce information
about a private key may be performed offline. The amount of resources
required for such an attack would be very significant and likely only
accessible to a limited number of attackers. An attacker would
additionally need online access to an unpatched system using the target
private key in a scenario with persistent DH parameters and a private
key that is shared between multiple clients.

This only affects processors that support the BMI1, BMI2 and ADX extensions like
Intel Broadwell (5th generation) and later or AMD Ryzen.

Note: This issue is very similar to CVE-2017-3732 and CVE-2015-3193 but must be
treated as a separate problem.

OpenSSL 1.1.0 users should upgrade to 1.1.0g
OpenSSL 1.0.2 users should upgrade to 1.0.2m

This issue was reported to OpenSSL on 10th August 2017 by the OSS-Fuzz project.
The fix was developed by Andy Polyakov of the OpenSSL development team.

`docker exec -it r-XXXXXXX-lb-1-387cf0f9 bash
root@58c42bca08d2:/# openssl version
OpenSSL 1.0.2g  1 Mar 2016`

Is Rancher going to provide a patch for this security vulnerability?

**Comment by deniseschannon** [comment]
@superseb can you test with ` lb-service-haproxy:v0.9.0`?

**Comment by superseb** [comment]
```
Ubuntu 16.04 LTS (Xenial Xerus): | released (1.0.2g-1ubuntu4.9)
```
Source: https://people.canonical.com/~ubuntu-security/cve/2017/CVE-2017-3736.html

```
$ docker run -t --entrypoint bash rancher/lb-service-haproxy:v0.9.0 -c "dpkg -l | grep openssl"
ii  openssl                     1.0.2g-1ubuntu4.10                    amd64        Secure Sockets Layer toolkit - cryptographic utility
```

@zvercodebender Let me know if you don't agree.
